addons/website_forum/models/forum.py

```diff
@@ -510,13 +510,14 @@ def _update_content(self, content, forum_id):
         if content and self.env.user.karma < forum.karma_dofollow:
             for match in re.findall(r'<a\s.*href=".*?">', content):
                 match = re.escape(match)  # replace parenthesis or special char in regex
```

Comment:
Instead of using a global "replace" here below, can't we match based on escaped string but replace with non-escaped version?

```py
escaped_match = re.escape(match)  # replace parenthesis or special char in regex
content = re.sub(escaped_match , match[:3] + 'rel="nofollow" ' + match[3:], content)
```

Also not sure what the "ugc" changes are about.

Reply:
Hello @awa-odoo 
Thanks for the review, as you mentioned in this code will work but
```py 
   escaped_match = re.escape(match)  # replace parenthesis or special char in regex
   content = re.sub(escaped_match , match[:3] + 'rel="nofollow" ' + match[3:], content) 
   ```

later performing a regular expression on content which might get the wrong value because of not doing escaped on content?

About the "ugc", it was already mentioned on 'rel' attribute inside 'a' tag and we were adding a new rel with nofollow which later got removed only rel="ugc" left, as mentioned below
```html 
	<a href="https://www%5C.youtube%5C.com/" rel="ugc">cc</a> 
```
so adding nofollow in the existing.
Correct me, if I am wrong!

Thank you...